src/app/utils/field-validation/validators/numberValidator.ts

```diff
@@ -0,0 +1,75 @@
+import { chain, left, right } from 'fp-ts/lib/Either'
+import { flow } from 'fp-ts/lib/function'
+
+import { ProcessedSingleAnswerResponse } from 'src/app/modules/submission/submission.types'
+import { INumberField } from 'src/types/field'
+import { ResponseValidator } from 'src/types/field/utils/validation'
+
+import { NumberSelectedValidation } from '../../../../types/field'
+
+import { notEmptySingleAnswerResponse } from './common'
+
+type NumberValidator = ResponseValidator<ProcessedSingleAnswerResponse>
+type NumberValidatorConstructor = (numberField: INumberField) => NumberValidator
+
+/**
+ * Validator, to check that answer is an empty string or a nonnegative integer
+ */
+const numberFormatValidator: NumberValidator = (response) => {
+  const { answer } = response
+  return /^\d*$/.test(answer)
+    ? right(response)
+    : left(`NumberValidator:\t answer is not a valid number format`)
+}
+
+const minLengthValidator: NumberValidatorConstructor = (numberField) => (
+  response,
+) => {
+  const { answer } = response
+  const { customMin } = numberField.ValidationOptions
+  return !customMin || answer.length >= customMin
+    ? right(response)
+    : left(`NumberValidator:\t answer is shorter than custom minimum length`)
+}
+
+const maxLengthValidator: NumberValidatorConstructor = (numberField) => (
+  response,
+) => {
+  const { answer } = response
+  const { customMax } = numberField.ValidationOptions
+  return !customMax || answer.length <= customMax
+    ? right(response)
+    : left(`NumberValidator:\t answer is longer than custom maximum length`)
+}
+
+const exactLengthValidator: NumberValidatorConstructor = (numberField) => (
+  response,
+) => {
+  const { answer } = response
+  const { customVal } = numberField.ValidationOptions
+  return !customVal || answer.length === customVal
+    ? right(response)
+    : left(`NumberValidator:\t answer does not match custom exact length`)
+}
+
+const getNumberLengthValidator: NumberValidatorConstructor = (numberField) => {
+  switch (numberField.ValidationOptions.selectedValidation) {
+    case NumberSelectedValidation.Min:
+      return minLengthValidator(numberField)
+    case NumberSelectedValidation.Max:
+      return maxLengthValidator(numberField)
+    case NumberSelectedValidation.Exact:
+      return exactLengthValidator(numberField)
+    default:
```

Comment:
~~Should there also be a `NumberSelectedValidation.Range` case? Seems like it was not in the old code but should also be handled here.~~

EDIT: Just checked the code; `.Range` is not used anywhere, maybe we can delete that enum actually, if it does not exist in the DB. 

EDIT 2: Seems like there already is an issue tracked at https://github.com/opengovsg/FormSG/issues/408

Reply:
Thanks! Discussed at syncup and we'll port over the exact implementation first so we don't change too many things at once, and deal with #408 later